lib/active_merchant/billing/gateways/quickbooks.rb

```diff
@@ -204,44 +198,36 @@ def post_data(data = {})
       end
 
       def headers(method, uri)
+        raise ArgumentError, "Invalid HTTP method: #{method}. Valid methods are :post and :get" unless [:post, :get].include?(method)
         request_uri = URI.parse(uri)
 
-        request_class = case method
-        when :post
-          Net::HTTP::Post
-        when :get
-          Net::HTTP::Get
-        else
-          raise ArgumentError, "Invalid HTTP method: #{method}. Valid methods are :post and :get"
-        end
+        # Following the guidelines from http://nouncer.com/oauth/authentication.html
+        oauth_parameters = {
+          oauth_nonce: generate_unique_id,
+          oauth_timestamp: Time.now.to_i.to_s,
+          oauth_signature_method: 'HMAC-SHA1',
+          oauth_version: "1.0",
+          oauth_consumer_key: @options[:consumer_key],
+          oauth_token: @options[:access_token]
+        }
 
-        # required for Ruby 1.9 backwards compatibilty
```

Comment:
This was needed to satisfy the gem's client helper interface. It's needed for the Signature Base String. Since we have the `method` param already, we can just use it directly and not have to construct any `URI`s.